pyqtgraph/graphicsItems/TargetItem.py

```diff
@@ -1,132 +1,429 @@
+        if dt is None:
+            self._shape = self._path
+            return None
+        v = dt.map(QtCore.QPointF(1, 0)) - dt.map(QtCore.QPointF(0, 0))
+        dti = fn.invertQTransform(dt)
+        devPos = dt.map(QtCore.QPointF(0, 0))
+        tr = QtGui.QTransform()
+        tr.translate(devPos.x(), devPos.y())
+        va = atan2(v.y(), v.x())
+        tr.rotate(va * 180.0 / pi)
+        tr.scale(self.scale, self.scale)
+        return dti.map(tr.map(self._path))
 
     def mouseDragEvent(self, ev):
-        if not self.movable:
+        if not self.movable or int(ev.button() & QtCore.Qt.LeftButton) == 0:
+            return
+        ev.accept()
+        if ev.isStart():
+            self.symbolOffset = self.pos() - self.mapToParent(ev.buttonDownPos())
+            self.moving = True
+
+        if not self.moving:
             return
-        if ev.button() == QtCore.Qt.LeftButton:
-            if ev.isStart():
-                self.moving = True
-                self.cursorOffset = self.pos() - self.mapToParent(ev.buttonDownPos())
-                self.startPosition = self.pos()
+        self.setPos(self.symbolOffset + self.mapToParent(ev.pos()))
+
+        if ev.isFinish():
+            self.moving = False
+            self.sigPositionChangeFinished.emit(self)
+
+    def mouseClickEvent(self, ev):
+        if self.moving and ev.button() == QtCore.Qt.RightButton:
             ev.accept()
-            
-            if not self.moving:
-                return
-                
-            self.setPos(self.cursorOffset + self.mapToParent(ev.pos()))
-            if ev.isFinish():
-                self.moving = False
-                self.sigDragged.emit(self)
+            self.moving = False
+            self.sigPositionChanged.emit(self)
+            self.sigPositionChangeFinished.emit(self)
+
+    def setMouseHover(self, hover):
+        # Inform the item that the mouse is(not) hovering over it
+        if self.mouseHovering is hover:
+            return
+        self.mouseHovering = hover
+        if hover:
+            self.currentBrush = self.hoverBrush
+            self.currentPen = self.hoverPen
+        else:
+            self.currentBrush = self.brush
+            self.currentPen = self.pen
+        self.update()
 
     def hoverEvent(self, ev):
-        if self.movable:
-            ev.acceptDrags(QtCore.Qt.LeftButton)
+        if self.movable and (not ev.isExit()) and ev.acceptDrags(QtCore.Qt.LeftButton):
+            self.setMouseHover(True)
+        else:
+            self.setMouseHover(False)
+
+    def viewTransformChanged(self):
+        GraphicsObject.viewTransformChanged(self)
+        self._shape = None  # invalidate shape, recompute later if requested.
+        self.update()
+
+    def pos(self):
+        """Provides the current position of the TargetItem
+
+        Returns
+        -------
+        tuple
+            ``(x, y)`` position of the TargetItem
+        """
+        return self._pos
```

Comment:
This should probably return a `Point`. Most other `pos` methods I found do as much, and when dragging, I get this error:
```
    |  Traceback (most recent call last):
    |    File "examples/InfiniteLine.py", line 86, in <module>
    |      QtGui.QApplication.instance().exec_()
    |    File "/home/martin/src/acq4/pyqtgraph/pyqtgraph/widgets/GraphicsView.py", line 372, in mouseMoveEvent
    |      super().mouseMoveEvent(ev)
    |    File "/home/martin/src/acq4/pyqtgraph/pyqtgraph/GraphicsScene/GraphicsScene.py", line 218, in mouseMoveEvent
    |      if self.sendDragEvent(ev, init=init):
    |    File "/home/martin/src/acq4/pyqtgraph/pyqtgraph/GraphicsScene/GraphicsScene.py", line 343, in sendDragEvent
    |      debug.printExc("Error sending drag event:")
    |    --- exception caught here ---
    |    File "/home/martin/src/acq4/pyqtgraph/pyqtgraph/GraphicsScene/GraphicsScene.py", line 341, in sendDragEvent
    |      item.mouseDragEvent(event)
    |    File "/home/martin/src/acq4/pyqtgraph/pyqtgraph/graphicsItems/TargetItem.py", line 231, in mouseDragEvent
    |      self.symbolOffset = self.pos() - self.mapToParent(ev.buttonDownPos())
    |  TypeError: unsupported operand type(s) for -: 'tuple' and 'QPointF'
```

Reply:
Ahh good point, should have checked what similar methods do elsewhere in the library.